ovirt/resource_ovirt_vm.go

```diff
@@ -376,6 +382,61 @@ func resourceOvirtVMCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 	vmBuilder.Cpu(cpu)
 
+	if blockDeviceOk {
+		if storage_domain, _ := blockDevice.([]interface{})[0].(map[string]interface{})["storage_domain"]; storage_domain != "" && templateIDOK {
+
+			// Get the reference to the service that manages the storage domains
+			sdsService := conn.SystemService().StorageDomainsService()
+
+			// Find the storage domain we want to be used for virtual machine disks
+			sdsResp, err := sdsService.List().Search("name=" + storage_domain.(string)).Send()
+			if err != nil {
+				return fmt.Errorf("Failed to search storage domains, reason: %v", err)
+			}
+
+			tds, err := getTemplateDiskAttachments(templateID.(string), meta)
+			if err != nil {
+				return fmt.Errorf("Failed to get Template disks attachments: %v", err)
+			}
+
+			if storageDomains, ok := sdsResp.StorageDomains(); ok {
+				sd := storageDomains.Slice()[0]
```

Comment:
This panics if a storage domain wasn't found - a protection like below should be added.

```
if len(storageDomains.Slice()) == 0 {
    fmt.Errorf("failed to find a storage domain with name=%s", name)
}
```

Reply:
Thank you. Protection was added.